flask/app/groups.py

```diff
@@ -170,43 +187,52 @@ def create_group():
     except:
```

Comment:
playing around with this and if the db constraint fails on group_code it will throw a 422 here

Reply:
I'm sorry, I'm not sure what the problem is here. The database constraint on `group_code` is a valid domain name, which matches the S3 requirements for a bucket. We try to make the bucket first to be confident before committing it to MySQL.